### PR TITLE
feat: Attachment title will be used to populate Draft Subject

### PR DESCRIPTION
**Pull request by adrien-coye**
*** conditions may apply, depends on type of underlying NSItemProvider.

### CODE REVIEW

File: Mail/Views/New Message/Attachments/Attachable.swift [hunk 1]
```diff
@@ -38,27 +38,32 @@ extension NSItemProvider: Attachable {
         return UTType(preferredIdentifier)
     }
 
-    public func writeToTemporaryURL() async throws -> URL {
+    public func writeToTemporaryURL() async throws -> (url: URL, title: String?) {
```
Comment (adrien-coye) [inline Mail/Views/New Message/Attachments/Attachable.swift:41]: maybe a marginal improvement using a typealias for this `(url: URL, title: String?)`

File: MailCore/Cache/AttachmentsManagerWorker/AttachmentsManagerWorker.swift [hunk 2]
```diff
@@ -88,6 +88,25 @@ public final class AttachmentsManagerWorker {
         liveAttachments.map { $0.detached() }
     }
 
+    var attachmentsTitles: [String?]? {
```
Comment (adrien-coye) [inline MailCore/Cache/AttachmentsManagerWorker/AttachmentsManagerWorker.swift:91]: maybe some doc

**Comment by adrien-coye** [comment]
I tested the feature on iPhone8 iOS16.7 all good. Removing draft status.

**Comment by PhilippeWeidmann** [comment]
I think users were expecting something more like this:
https://stackoverflow.com/a/33139355/9066648
However this is requires a lot more code so I'm okay to first merge this.

**Comment by adrien-coye** [comment]
I tried to match the behaviour of Android that does not just work for webpages, therefore I did what was to me the most pragmatic first step. The good thing is that what you are sharing can be added later @PhilippeWeidmann

**Pull request merged**